### DISCUSSION TITLE
Using multiple custom validations on one field

### DISCUSSION THREAD

**Discussion by alexcstaley** [go-playground/validator#1355] [Q&A]
Hello, curious if this is supported, e.g.
```
mySchema = map[string]string{
  "myField": "custom_tag_1,custom_tag_2",
}

schema := v.New()
schema.RegisterStructValidationMapRules(mySchema, structOfSomeKnownType)
schema.RegisterValidation("custom_tag_1", CustomValidationFunc1(), true)
schema.RegisterValidation("custom_tag_2", CustomValidationFunc2(), true)
```

The above seems to register both my custom rules successfully, but as far as I can tell, only the rule associated with the tag coming before the comma (`custom_tag_1`) is actually executed (if I switch the order in the rules map to `"myField": "custom_tag_2,custom_tag_1",`, then only `CustomValidationFunc2()` is executed). Am I obviously using this incorrectly, or is this behavior not supported? Thanks in advance 🙏

**Comment by nodivbyzero** [comment]
How about calling `registerValidation()` before `RegisterStructValidationMapRules`?
```
func validateStruct() {
	data := Data2{
		Name: "leo1",
		Age:  5,
	}

	rules := map[string]string{
		"Name": "min=4,max=6,fooValidation,barValidation",
		"Age":  "min=4,max=6",
	}
	validate.RegisterValidation("fooValidation", ValidateFoo)
	validate.RegisterValidation("barValidation", ValidateBar)
	validate.RegisterStructValidationMapRules(rules, Data2{})

	err := validate.Struct(data)
	fmt.Println(err)
	fmt.Println()
}

func ValidateFoo(fl validator.FieldLevel) bool {
	fmt.Printf("ValidateFoo %v\n", fl.FieldName())
	return true
}

func ValidateBar(fl validator.FieldLevel) bool {
	fmt.Printf("ValidateBar %v\n", fl.FieldName())
	return true
}

output:
ValidateFoo Name
ValidateBar Name
<nil>

```

**Reply by alexcstaley** [reply]
Thanks for the suggestion 🙏 
Calling `RegisterValidation()` before `RegisterStructValidationMapRules()` doesn't seem to have an effect. My change is on the urgent side of things so I ended up just combining my two validation cases into one custom validation, sacrificing a bit of rejections message precision to get it done. So I probably won't reimplement.

But now that I'm testing fixes I'm finding some interesting behavior... I'm flipping my two custom validations back and forth in the rule map, `"myField": "custom_tag_1,custom_tag_2"` <--> `"myField": "custom_tag_2,custom_tag_1"`, and it's calling both validation functions for one order but not the other (this is not what I was seeing this morning). So I don't know, maybe there's some weird race condition happening, or (more likely) I just had/have some stupid mistake in my code somewhere. But on the off-chance I've discovered a feature gap I encourage others to try and replicate this issue, and welcome their findings

**Reply by nodivbyzero** [reply]
I can confirm that the order in which `RegisterValidation` and `RegisterStructValidationMapRules` are called does not matter. 

I tested this using the example I shared in my earlier comment.
